diff --git a/conf/dionaea.conf.dist b/conf/dionaea.conf.dist
index 43e17b4..7b16529 100644
--- a/conf/dionaea.conf.dist
+++ b/conf/dionaea.conf.dist
@@ -344,6 +344,16 @@ modules = {
 			user = "" 		// username (optional)
 			pass = ""		// password (optional)
 		}
+		hpfeeds = {
+			hp1 = {
+				server = ""
+				port = ""
+				ident = ""
+				secret = ""
+				// dynip_resolve: enable to lookup the sensor ip through a webservice
+				dynip_resolve = "http://hpfeeds.honeycloud.net/ip"
+			}
+		}
 		logsql = {
 			mode = "sqlite" // so far there is only sqlite
 			sqlite = {
@@ -457,6 +467,7 @@ modules = {
 //			"virustotal",
 //			"mwserv",
 //			"submit_http",
+			"hpfeeds",
 //			"logxmpp",
 //			"nfq",
 //			"p0f",
diff --git a/modules/python/scripts/ihandlers.py b/modules/python/scripts/ihandlers.py
index 5b40dd7..8bddc63 100644
--- a/modules/python/scripts/ihandlers.py
+++ b/modules/python/scripts/ihandlers.py
@@ -129,6 +129,13 @@ def new():
 		import dionaea.submit_http
 		g_handlers.append(dionaea.submit_http.handler('*'))
 
+	if "hpfeeds" in g_dionaea.config()['modules']['python']['ihandlers']['handlers'] and 'hpfeeds' in g_dionaea.config()['modules']['python']:
+		import dionaea.hpfeeds
+		for client in g_dionaea.config()['modules']['python']['hpfeeds']:
+			conf = g_dionaea.config()['modules']['python']['hpfeeds'][client]
+			x = dionaea.hpfeeds.hpfeedihandler(conf)
+			g_handlers.append(x)
+
 	if "fail2ban" in g_dionaea.config()['modules']['python']['ihandlers']['handlers']:
 		import dionaea.fail2ban
 		g_handlers.append(dionaea.fail2ban.fail2banhandler())
